app/src/main/java/com/epfl/beatlink/model/map/user/MapUsersViewModel.kt

```diff
@@ -0,0 +1,138 @@
+package com.epfl.beatlink.model.map.user
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.epfl.beatlink.model.spotify.objects.SpotifyAlbum
+import com.epfl.beatlink.model.spotify.objects.SpotifyArtist
+import com.epfl.beatlink.model.spotify.objects.SpotifyTrack
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+open class MapUsersViewModel(private val repository: MapUsersRepository) : ViewModel() {
+
+  // Private mutable LiveData to hold the data
+  private val _mapUsers = MutableStateFlow<List<MapUser>>(emptyList())
+  private var _mapUser = MutableStateFlow<MapUser?>(null)
+  private val _authState = MutableStateFlow<Boolean>(false)
+  private val _playbackState = MutableStateFlow<CurrentPlayingTrack?>(null)
+  private val _playbackStateFetched = MutableStateFlow<Boolean>(false)
+
+  // Public immutable StateFlows for observers
+  val mapUsers: StateFlow<List<MapUser>> = _mapUsers.asStateFlow()
+  val mapUser: StateFlow<MapUser?> = _mapUser.asStateFlow()
+  val authState: StateFlow<Boolean> = _authState.asStateFlow()
+  val playbackState: StateFlow<CurrentPlayingTrack?> = _playbackState.asStateFlow()
+  val playbackStateFetched: StateFlow<Boolean> = _playbackStateFetched.asStateFlow()
+
+  init {
+    // Initialize repository and listen to authentication state
+    repository.init {
+      _authState.value = true
+      Log.d("TRUE", "TRUE")
```

Comment:
You might wanna change this, as it doesn't explain anything. If you used for testing purpose just delete them, otherwise make it more explicit.